src/coreclr/src/jit/lsraarm64.cpp

```diff
@@ -1050,9 +1050,24 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
 
     if (intrin.op1 != nullptr)
     {
-        // If we have an RMW intrinsic, we want to preference op1Reg to the target if
-        // op1 is not contained.
-        if (isRMW)
+        bool simdRegToSimdRegMove = false;
+
+        if ((intrin.id == NI_Vector64_CreateScalarUnsafe) || (intrin.id == NI_Vector128_CreateScalarUnsafe))
+        {
+            simdRegToSimdRegMove = varTypeIsFloating(intrin.op1);
+        }
+        else if (intrin.id == NI_AdvSimd_Arm64_DuplicateToVector64)
+        {
+            simdRegToSimdRegMove = (intrin.op1->TypeGet() == TYP_DOUBLE);
+        }
+        else if ((intrin.id == NI_Vector64_ToScalar) || (intrin.id == NI_Vector128_ToScalar))
```

Comment:
>intrin.id == NI_Vector64_ToScalar [](start = 18, length = 33)

should we also special case for `GetElement(0)`

Reply:
Yes, we could this for `GetElement(0)` and some others (e.g. `AdvSimd.Extract` with `0` index). 
I decided not to do optimization part at the moment and keep such changes for .NET 6.0.

The reason why I changed LSRA in this PR - because treating `MathF.FusedMultiplyAdd` as an intrincis introduced the annoying and unnecessary `fmov` (as you can see in the codegen diffs) and I wanted to avoid that.

Reply:
Would it make sense to open an issue for .NET 6 to ensure that we are doing this for all intrinsics that have move/copy semantics?

Reply:
There are likely a lot of places where we could introduce simple optimizations for things like extractions or shuffles (on both x86 and ARM).

LLVM/MSVC do many of these already and it might be good to construct a general list of them for potential .NET 6 work